meta/Makefile

```diff
@@ -68,6 +68,14 @@ $(foreach bin,$(BINS),$(if $(shell which $(bin)),,$(error "Missing $(bin) in PAT
 
 CFLAGS += -I../inc -I../experimental $(WARNINGS)
 
+ifeq ($(CROSS_BUILD_ENVIRON),y)
```

Comment:
CROSS_BUILD_ENVIRON is that your internal variable name?

Reply:
This environment variable is set in Sonic slave.mk when Sonic build uses cross-complation.

Reply:
i grepped sonic-buildimage and i could not find it, can you give a link in github where this is set up?

Reply:
also, im just wondering, normally when we have configure.ac and ./configure, to specify cross compiler paramst are used  from --host=HOST, this is standalone Makefile, so maybe it make more sense to make it similar like here: https://lists.gnu.org/archive/html/help-make/2005-05/msg00029.html and then passing 
```
$ CROSS_COMPILE=xxx make
```
or whoever would set CROSS_COMPILE, or it could be even set CC and CXX before make, instead of doing all those if/else across all makefiles on the entire sonic

Reply:
This sonic-buildimage PR https://github.com/Azure/sonic-buildimage/pull/8035 is part of of the overall PR. You can find CROSS_BUILD_ENVIRON environment variable set in slave.mk that I've modified. The naming convention is taken from another variable MULTIARCH_QEMU_ENVIRON defined in this file. CROSS_BUILD_ENVIRON env. variable is used in hundreds files in the PR. I'll check the rest of your request tomorrow. Thanks for your comments.